### PR TITLE
ci(test): run typechecking before jest

### PR DESCRIPTION
**Pull request by HorusGoul**
This adds a new step for the test job to ensure that all the typescript code is correct before running the tests.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -154,3 +151,41 @@ jobs:
...
+            */*/node_modules
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
+
+      - name: Install dependencies
+        run: yarn
+
+      - name: Run typechecking
+        run: yarn typecheck
```
Comment (mpeyper) [inline .github/workflows/test.yml]: should it run `lint` as well?

**Comment by mpeyper** [comment]
should `yarn typecheck` also be run in `husky pre-commit`/`lint-staged`?

File: .github/workflows/test.yml (hunk 1 above)
Comment (HorusGoul) [reply .github/workflows/test.yml]: Let's create another job for the linter!

**Comment by HorusGoul** [comment]
@mpeyper Not necessary, type checking is usually slow, and I don't feel like we should make people wait every time they create a new commit.

**Pull request merged**